app/ide-desktop/lib/common/src/index.ts

```diff
@@ -5,6 +5,26 @@
  * here when it is not possible for a sibling package to own that code without introducing a
  * circular dependency in our packages. */
 
+// ================
+// === Platform ===
+// ================
+
+/** Defines the platform the application is running on.
+ *
+ * Depending on the platform, the application will use different routing mechanisms. For example, in
+ * the cloud, the application will use the browser's URL bar to navigate between pages. In Electron,
+ * the application will use the `MemoryRouter` to navigate between pages. Similarly, the
+ * application will use different redirect URLs for authentication. For example, in the cloud, the
+ * application will redirect using `http://` URLs. In Electron, the application will redirect using
+ * `enso://` URLs. The former flow works entirely in-browser. The latter flow must go to the browser
+ * and back to the application. */
+export enum Platform {
+    /** Application is running on a desktop (i.e., in Electron). */
+    desktop = 'desktop',
+    /** Application is running in the browser (i.e., in the cloud). */
+    cloud = 'cloud',
```

Comment:
This is incompatible with Rust. In Rust we support platform "web", "macOS", "window", etc. We don't support "desktop", so this could throw strange things in Rust code sometimes.

Reply:
this is for the dashboard only - the dashboard cannot access the IDE options, so it needs its own platform enum. maybe the naming can be improved though

Reply:
maybe it should be removed and replaced with something like `supportsLocalBackend` instead?